transformers/tokenization_utils.py

```diff
@@ -1085,7 +1085,7 @@ def decode(self, token_ids, skip_special_tokens=False, clean_up_tokenization_spa
                 if current_sub_text:
                     sub_texts.append(self.convert_tokens_to_string(current_sub_text))
                     current_sub_text = []
-                sub_texts.append(" " + token)
+                sub_texts.append(" " + token + " ")
```

Comment:
Wouldn't this generate double spaces in the output string?

Reply:
Actually, that's what I thought at start but no. Text tokens are joined with spaces in `convert_tokens_to_string` but then subtexts are joined with `''` and a space is missing after added tokens. Maybe my correction is too superficial and a rethink of this decoder formatting and join with spaces (or another separator) would be needed.